### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by anetta999**
## Describe your changes

- created signin/signup pages
- created signin/signup forms

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://olga-team.atlassian.net/browse/TP-50, https://olga-team.atlassian.net/browse/TP-56)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [ ] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit f

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### Screenshots of the changes
image
image
image

image
image
image

### CODE REVIEW

File: src/components/signInForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  const onSubmit: SubmitHandler<Inputs> = async data => {
+    try {
+      const response = await login(data).unwrap();
+      if (!response?.is_verified) {
+        await sendOtp(data.email);
+        reset();
+        navigate('');
+      } else {
+        dispatch(setToken(response?.access_token));
+        navigate('/');
```
Comment (MishaFomenko) [inline src/components/signInForm/index.tsx]: should not be here

File: src/components/signInForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            className={errors.password ? 'error-input' : ''}
+            placeholder={t('signIn.passwordPlaceholder')}
+            id="user-password"
+          />
+          {errors.password && <ErrorMsg>{errors.password.message}</ErrorMsg>}
+          <button type="button" onClick={handlePassword}>
+            {isPasswordShown ? <EyeIcon /> : <EyeCloseIcon />}
+          </button>
+        </div>
+        <PasswordLink to="/restore-password">
```
Comment (MishaFomenko) [inline src/components/signInForm/index.tsx]: link

File: src/components/signInForm/signInHook.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const signInSchema = yup
+    .object()
+    .shape({
+      email: yup
+        .string()
+        .email(t('validation.email'))
+        .required('Please enter your credentials'),
+      password: yup
+        .string()
+        .min(8, 'Password format is incorrect')
```
Comment (MishaFomenko) [inline src/components/signInForm/signInHook.ts]: strings and magic numbers should be extracted

File: src/components/signUpForm/signUpFormHooks.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        .min(8, t('validation.password'))
+        .required(t('validation.credentials')),
+      repeatedPassword: yup
+        .string()
+        .test(
+          'isRepeatedPasswordValueMatched',
+          t('validation.repeatedPassword'),
+          (value, { parent }) => !value || value === parent.password,
+        )
+        .min(8, t('validation.password'))
```
Comment (MishaFomenko) [inline src/components/signUpForm/signUpFormHooks.ts]: the number might have to be a variable

File: src/pages/signInPage/index.tsx [hunk 5]
```diff
@@ -1,5 +1,64 @@
...
+import { FormContainer } from 'src/pages/signUpPage/styles.ts';
+import bgImg from '../../assets/images/sign-in.png';
+import { ArrowBackLink } from 'components/arrowBackLink';
+
 const SignInPage = () => {
-  return <div></div>;
+  const { t } = useTranslation();
+
+  const verifyGoogle = async () => {
+    window.location.href = 'http://localhost:3000/google';
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx]: link

File: src/pages/signInPage/index.tsx [hunk 6]
```diff
@@ -1,5 +1,64 @@
...
+    <SignUpSection>
+      <LogoContainer img={bgImg}>
+        <LogoWrap>
+          <LogoIcon width={203} height={56} />
+        </LogoWrap>
+      </LogoContainer>
+      <FormContainer>
+        <FormHeader>
+          <Title>
+            <ArrowBackLink link={'/faq_team_fe/signin'} />
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx]: link

File: src/pages/signInPage/index.tsx [hunk 7]
```diff
@@ -1,5 +1,64 @@
...
+          </Title>
+          <SubTitle>{t('signIn.subtitle')}</SubTitle>
+          <Google type="button" onClick={verifyGoogle}>
+            <GoogleIcon />
+            {t('signIn.google')}
+          </Google>
+          <Text>or</Text>
+        </FormHeader>
+        <SignInForm />
+        <SignUpLink to="/signup">
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx]: link

File: src/pages/signInPage/index.tsx [hunk 8]
```diff
@@ -1,5 +1,64 @@
...
+          <p>{t('signIn.signUpLink')}</p>
+          <span>{t('signIn.signUp')}</span>
+        </SignUpLink>
+        <ListContainer>
+          <li>
+            <PolicyLink to="/privacy-policy">
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx]: link

File: src/pages/signInPage/index.tsx [hunk 9]
```diff
@@ -1,5 +1,64 @@
...
+              {t('signIn.policyLink')}
+            </PolicyLink>
+          </li>
+          <li>
+            <PolicyLink to="/terms-of-use">{t('signIn.termsLink')}</PolicyLink>
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx]: lin

File: src/pages/signInPage/style.ts [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    @media screen and (min-width: ${({ theme }) => theme.breakpoint.desktop}) {
+      background-color: ${({ theme }) => theme.colors.border};
+    }
+  }
+`;
+
+export const SignInSection = styled(Section)<{ theme?: Theme }>`
+  @media screen and (max-width: 767px) {
```
Comment (MishaFomenko) [inline src/pages/signInPage/style.ts]: screen width should be a constant

File: src/pages/signUpPage/index.tsx [hunk 11]
```diff
@@ -1,5 +1,69 @@
...
+  SignUpLink,
+} from '../signInPage/style';
+import bgImg from '../../assets/images/sign-up.png';
+import { ArrowBackLink } from 'components/arrowBackLink';
 const SignUpPage = () => {
-  return <div></div>;
+  const { t } = useTranslation();
+
+  const verifyGoogle = async () => {
+    window.location.href = 'http://localhost:3000/google';
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: link

File: src/pages/signUpPage/index.tsx [hunk 12]
```diff
@@ -1,5 +1,69 @@
...
+    <SignUpSection>
+      <LogoContainer img={bgImg}>
+        <LogoWrap>
+          <LogoIcon width={203} height={56} />
+        </LogoWrap>
+      </LogoContainer>
+      <FormContainer>
+        <FormHeader>
+          <Title>
+            <ArrowBackLink link={'/faq_team_fe/signin'} />
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: link

File: src/pages/signUpPage/index.tsx [hunk 13]
```diff
@@ -1,5 +1,69 @@
...
+          </Title>
+          <SubTitle>{t('signUp.subtitle')}</SubTitle>
+          <Google type="button" onClick={verifyGoogle}>
+            <GoogleIcon />
+            {t('signUp.google')}
+          </Google>
+          <Text>or</Text>
+        </FormHeader>
+        <SignUpForm />
+        <SignUpLink to="/signin">
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: link

File: src/pages/signUpPage/index.tsx [hunk 14]
```diff
@@ -1,5 +1,69 @@
...
+          <p>{t('signUp.signInLink')}</p>
+          <span>{t('signUp.signIn')}</span>
+        </SignUpLink>
+        <ListContainer>
+          <li>
+            <PolicyLink to="/privacy-policy">
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: link

File: src/pages/signUpPage/index.tsx [hunk 15]
```diff
@@ -1,5 +1,69 @@
...
+              {t('signUp.policyLink')}
+            </PolicyLink>
+          </li>
+          <li>
+            <PolicyLink to="/terms-of-use">{t('signUp.termsLink')}</PolicyLink>
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: l

File: src/components/signInForm/index.tsx (hunk 1 above)
Comment (anetta999) [reply src/components/signInForm/index.tsx]: fixed

File: src/components/signInForm/index.tsx (hunk 2 above)
Comment (anetta999) [reply src/components/signInForm/index.tsx]: fixed

File: src/components/signInForm/signInHook.ts (hunk 3 above)
Comment (anetta999) [reply src/components/signInForm/signInHook.ts]: fixed

File: src/components/signUpForm/signUpFormHooks.ts (hunk 4 above)
Comment (anetta999) [reply src/components/signUpForm/signUpFormHooks.ts]: fixed

File: src/pages/signInPage/index.tsx (hunk 5 above)
Comment (anetta999) [reply src/pages/signInPage/index.tsx]: fixed

File: src/pages/signInPage/index.tsx (hunk 6 above)
Comment (anetta999) [reply src/pages/signInPage/index.tsx]: fixed

File: src/pages/signInPage/index.tsx (hunk 7 above)
Comment (anetta999) [reply src/pages/signInPage/index.tsx]: fixed

File: src/pages/signInPage/index.tsx (hunk 8 above)
Comment (anetta999) [reply src/pages/signInPage/index.tsx]: fixed

File: src/pages/signInPage/index.tsx (hunk 9 above)
Comment (anetta999) [reply src/pages/signInPage/index.tsx]: fixed

File: src/pages/signInPage/style.ts (hunk 10 above)
Comment (anetta999) [reply src/pages/signInPage/style.ts]: fixed

File: src/pages/signUpPage/index.tsx (hunk 11 above)
Comment (anetta999) [reply src/pages/signUpPage/index.tsx]: fixed

File: src/pages/signUpPage/index.tsx (hunk 12 above)
Comment (anetta999) [reply src/pages/signUpPage/index.tsx]: fixed

File: src/pages/signUpPage/index.tsx (hunk 13 above)
Comment (anetta999) [reply src/pages/signUpPage/index.tsx]: fixed

File: src/pages/signUpPage/index.tsx (hunk 14 above)
Comment (anetta999) [reply src/pages/signUpPage/index.tsx]: fixed

File: src/pages/signUpPage/index.tsx (hunk 15 above)
Comment (anetta999) [reply src/pages/signUpPage/index.tsx]: fixed

File: src/components/signInForm/index.tsx [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+  useRestorePassMutation,
+} from 'src/redux/authApiSlice';
+import { useAppDispatch } from 'src/redux/hooks';
+import { setToken } from 'src/redux/auth/authSlice';
+import { useNavigate } from 'react-router-dom';
+import { useSignInSchema } from './signInFormHook';
+import {
+  isErrorWithMessage,
+  isFetchBaseQueryError,
+} from '../../helpers/errorHandler';
```
Comment (lyudad) [inline src/components/signInForm/index.tsx]: alias should be used

File: src/components/signInForm/index.tsx [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+
+export const SignInForm = () => {
+  const verifyRoute: string = '/';
+  const passwordRoute: string = '/restore-password';
```
Comment (lyudad) [inline src/components/signInForm/index.tsx]: move constant above the component, in other case they will reinitialise with each rerender

File: src/components/signUpForm/index.tsx [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+import EyeCloseIcon from 'src/assets/icons/iconEyeClose';
+import { useRegistrationMutation } from 'src/redux/authApiSlice';
+import { useNavigate } from 'react-router-dom';
+import { useAppDispatch } from 'src/redux/hooks.ts';
+import { setEmail } from 'src/redux/auth/authSlice.ts';
+import { useSignUpSchema } from './signUpFormHooks';
+import {
+  isErrorWithMessage,
+  isFetchBaseQueryError,
+} from '../../helpers/errorHandler';
```
Comment (lyudad) [inline src/components/signUpForm/index.tsx]: same

File: src/pages/signInPage/style.ts [hunk 19]
```diff
@@ -0,0 +1,224 @@
+import styled from '@emotion/styled';
+import { Theme } from 'src/styles/Theme';
+import { Link } from 'react-router-dom';
+import { Section } from 'src/components/section';
+import bgImg from '../../assets/images/sign-in.png';
```
Comment (lyudad) [inline src/pages/signInPage/style.ts]: same

File: src/pages/signInPage/style.ts (hunk 10 above)
Comment (lyudad) [reply src/pages/signInPage/style.ts]: look like no

File: src/pages/signUpPage/index.tsx (hunk 11 above)
Comment (lyudad) [reply src/pages/signUpPage/index.tsx]: not fixed

File: src/pages/signUpPage/index.tsx (hunk 12 above)
Comment (lyudad) [reply src/pages/signUpPage/index.tsx]: not fixed

File: src/components/signInForm/index.tsx (hunk 16 above)
Comment (anetta999) [reply src/components/signInForm/index.tsx]: corrected in here https://github.com/ZenBit-Tech/faq_team_fe/pull/5/commits/bf58e21075e7ea2cbbb782951c2912de63f79f65#diff-e8d970e863c84aacdbb71ce91fbc891b1cc898246e59a71a471b1edfae921b35

File: src/components/signInForm/index.tsx (hunk 17 above)
Comment (anetta999) [reply src/components/signInForm/index.tsx]: corrected in here https://github.com/ZenBit-Tech/faq_team_fe/pull/5/commits/bf58e21075e7ea2cbbb782951c2912de63f79f65#diff-e8d970e863c84aacdbb71ce91fbc891b1cc898246e59a71a471b1edfae921b35

File: src/components/signUpForm/index.tsx (hunk 18 above)
Comment (anetta999) [reply src/components/signUpForm/index.tsx]: corrected here https://github.com/ZenBit-Tech/faq_team_fe/pull/5/commits/bf58e21075e7ea2cbbb782951c2912de63f79f65#diff-9a808bbf778cf47d4daf721d3c55b0df1b53fa0c7f548c115aa5d73b33f832b4

File: src/pages/signInPage/style.ts (hunk 19 above)
Comment (anetta999) [reply src/pages/signInPage/style.ts]: corrected here https://github.com/ZenBit-Tech/faq_team_fe/pull/5/commits/bf58e21075e7ea2cbbb782951c2912de63f79f65#diff-34d6f7ccd03d49520dd3201e19bed4877a27d62929628c52b41d697334a247ea

File: src/pages/signUpPage/index.tsx [hunk 20]
```diff
@@ -1,5 +1,74 @@
...
+  FormContainer,
+} from './styles';
+import {
+  LogoWrap,
+  Title,
+  SubTitle,
+  Google,
+  Text,
+  SignUpLink,
+} from '../signInPage/styles';
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: needs an alias

File: src/pages/signUpPage/index.tsx [hunk 21]
```diff
@@ -1,5 +1,74 @@
+import { useTranslation } from 'react-i18next';
+import {
+  SignUpSection,
+  PolicyLink,
+  LogoContainer,
+  ListContainer,
+  FormHeader,
+  FormContainer,
+} from './styles';
```
Comment (MishaFomenko) [inline src/pages/signUpPage/index.tsx]: needs an alias

File: src/pages/signInPage/styles.ts [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    @media screen and (min-width: ${({ theme }) => theme.breakpoint.desktop}) {
+      background-color: ${({ theme }) => theme.colors.border};
+    }
+  }
+`;
+
+export const SignInSection = styled(Section)<{ theme?: Theme }>`
+  @media screen and (max-width: 767px) {
```
Comment (MishaFomenko) [inline src/pages/signInPage/styles.ts]: breakpoint needs to be a constant

File: src/pages/signInPage/styles.ts [hunk 23]
```diff
@@ -0,0 +1,224 @@
+import styled from '@emotion/styled';
+import { Link } from 'react-router-dom';
+import bgImg from '../../assets/images/sign-in.png';
```
Comment (MishaFomenko) [inline src/pages/signInPage/styles.ts]: alias

File: src/pages/signInPage/index.tsx [hunk 24]
```diff
@@ -1,5 +1,68 @@
+import { useTranslation } from 'react-i18next';
+import { SignUpLink, SubTitle, Title, Google, Text, LogoWrap } from './styles';
```
Comment (MishaFomenko) [inline src/pages/signInPage/index.tsx:2]: alias

File: src/helpers/errorHandler.ts [hunk 25]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (MishaFomenko) [inline src/helpers/errorHandler.ts]: commentar

File: src/components/signUpForm/index.tsx [hunk 26]
```diff
@@ -0,0 +1,145 @@
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useTranslation } from 'react-i18next';
+import { Inputs } from './types';
+import { StyledForm, SubmitBtn, ErrorMsg } from '../signInForm/styles';
```
Comment (MishaFomenko) [inline src/components/signUpForm/index.tsx]: alias

File: src/components/signInForm/styles.ts [hunk 27]
```diff
@@ -0,0 +1,146 @@
+import styled from '@emotion/styled';
+import { Link } from 'react-router-dom';
+import { Button } from '../button';
```
Comment (MishaFomenko) [inline src/components/signInForm/styles.ts]: alias

**Review by lyudad — APPROVED** [review]

**Pull request merged**